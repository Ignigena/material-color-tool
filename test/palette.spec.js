const { makePalette } = require('../')

it('uses the correct Material palette when appropriate', () => {
  expect([
    makePalette('#F44336'), // Red
    makePalette('#E91E63'), // Pink
    makePalette('#9C27B0'), // Purple
    makePalette('#673AB7'), // Deep Purple
    makePalette('#3F51B5'), // Indigo
    makePalette('#2196F3'), // Blue
    makePalette('#03A9F4'), // Light Blue
    makePalette('#00BCD4'), // Cyan
    makePalette('#009688'), // Teal
    makePalette('#4CAF50'), // Green
    makePalette('#8BC34A'), // Light Green
    makePalette('#CDDC39'), // Lime
    makePalette('#FFEB3B'), // Yellow
    makePalette('#FFC107'), // Amber
    makePalette('#FF9800'), // Orange
    makePalette('#FF5722'), // Deep Orange
    makePalette('#795548'), // Brown
    makePalette('#9E9E9E'), // Gray
    makePalette('#607D8B') // Blue Gray
  ]).toMatchSnapshot('Material')
})

it('generates spec-compliant palettes from custom colors', () => {
  expect([
    makePalette('#1E242D'), // Dark
    makePalette('#4A4A4A'), // Dark
    makePalette('#EFF0EF') // Light
  ]).toMatchSnapshot('Dark and Light')

  expect([
    makePalette('#E82127'), // Tesla "Lust"
    makePalette('#004225') // British Racing Green
  ]).toMatchSnapshot('Car Colors')

  expect([
    makePalette('#4285F4'),
    makePalette('#DB4437'),
    makePalette('#F4B400'),
    makePalette('#0F9D58')
  ]).toMatchSnapshot('Google Logo')
})

it('allows either a HEX code or HSV to be used', () => {
  expect(
    makePalette([358, 85.77586206896551, 90.98039215686275])
  ).toStrictEqual(
    makePalette('#E82127')
  )
})
