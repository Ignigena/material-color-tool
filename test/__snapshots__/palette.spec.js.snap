// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates spec-compliant palettes from custom colors: Car Colors 1`] = `
Array [
  Array [
    "#FFEAED",
    "#FFCACF",
    "#F29494",
    "#E96A6B",
    "#F24545",
    "#F62E27",
    "#E82128",
    "#D61122",
    "#C9011A",
    "#BB000B",
  ],
  Array [
    "#DFF0EB",
    "#B1DBCC",
    "#80C3AC",
    "#51AC8C",
    "#319A77",
    "#1A8963",
    "#167C58",
    "#116D4B",
    "#095D3E",
    "#004225",
  ],
]
`;

exports[`generates spec-compliant palettes from custom colors: Dark and Light 1`] = `
Array [
  Array [
    "#EBECF0",
    "#CCD1D9",
    "#ACB3BE",
    "#8B95A5",
    "#737F91",
    "#5C6A7F",
    "#4F5C6F",
    "#3F4A59",
    "#303844",
    "#1E242D",
  ],
  Array [
    "#FAFBFA",
    "#F4F5F4",
    "#EFF0EF",
    "#E2E3E2",
    "#BFC0BF",
    "#A0A1A0",
    "#777877",
    "#636463",
    "#444544",
    "#232323",
  ],
]
`;

exports[`generates spec-compliant palettes from custom colors: Google Logo 1`] = `
Array [
  Array [
    "#E4F2FF",
    "#BFDEFF",
    "#96CAFF",
    "#6DB5FF",
    "#53A4FF",
    "#4295FF",
    "#4286F4",
    "#3F74E0",
    "#3D62CD",
    "#3842AD",
  ],
  Array [
    "#FDEBEE",
    "#FBCED1",
    "#E99C9A",
    "#DD7773",
    "#E65B51",
    "#E94D38",
    "#DB4537",
    "#C93C31",
    "#BC352B",
    "#AC2D20",
  ],
  Array [
    "#FDF8E1",
    "#FBECB3",
    "#F8E082",
    "#F6D550",
    "#F5CA29",
    "#F4C10B",
    "#F4B300",
    "#F4A000",
    "#F49000",
    "#F47200",
  ],
  Array [
    "#E5F5EB",
    "#C0E5CE",
    "#98D4AF",
    "#6CC590",
    "#48B879",
    "#19AC62",
    "#0F9D58",
    "#018B4C",
    "#007A41",
    "#005B2D",
  ],
]
`;

exports[`uses the correct Material palette when appropriate: Material 1`] = `
Array [
  Array [
    "#FFEBEE",
    "#FFCDD2",
    "#EF9A9A",
    "#E57373",
    "#EF5350",
    "#F44336",
    "#E53935",
    "#D32F2F",
    "#C62828",
    "#B71B1C",
  ],
  Array [
    "#FCE4EC",
    "#F8BBD0",
    "#F48FB0",
    "#F06291",
    "#EC4079",
    "#E91E62",
    "#D81B5F",
    "#C2185A",
    "#AD1356",
    "#880D4E",
  ],
  Array [
    "#F3E5F5",
    "#E1BEE7",
    "#CE93D8",
    "#BA68C8",
    "#AB47BC",
    "#9B27B0",
    "#8D24AA",
    "#7A1FA2",
    "#691B9A",
    "#49148C",
  ],
  Array [
    "#EDE7F6",
    "#D1C4E9",
    "#B39DDB",
    "#9675CD",
    "#7F57C2",
    "#683AB7",
    "#5F35B1",
    "#522DA8",
    "#4627A0",
    "#321B92",
  ],
  Array [
    "#E8EAF6",
    "#C5CAE9",
    "#9FA8DA",
    "#7986CB",
    "#5C6BC0",
    "#3F51B5",
    "#3949AB",
    "#303F9F",
    "#283593",
    "#1A237E",
  ],
  Array [
    "#E3F2FD",
    "#BBDEFB",
    "#90C9F9",
    "#63B4F6",
    "#42A4F5",
    "#2194F3",
    "#1F87E5",
    "#1A75D2",
    "#1764C0",
    "#1045A1",
  ],
  Array [
    "#E1F5FE",
    "#B3E5FC",
    "#81D3FA",
    "#4EC2F7",
    "#28B5F6",
    "#03A8F4",
    "#049AE5",
    "#0487D1",
    "#0476BD",
    "#03569B",
  ],
  Array [
    "#E0F7FA",
    "#B2EBF2",
    "#80DEEA",
    "#4DCFE1",
    "#26C5DA",
    "#00BBD4",
    "#00ABC1",
    "#0096A7",
    "#00828F",
    "#005F64",
  ],
  Array [
    "#E0F2F1",
    "#B2DFDB",
    "#80CBC3",
    "#4DB6AB",
    "#26A699",
    "#009687",
    "#00897A",
    "#00796A",
    "#00695B",
    "#004D3F",
  ],
  Array [
    "#E8F5E9",
    "#C8E6C9",
    "#A5D6A7",
    "#81C784",
    "#66BB69",
    "#4CAF4F",
    "#43A046",
    "#388E3B",
    "#2E7D31",
    "#1B5E1F",
  ],
  Array [
    "#F1F8E9",
    "#DCEDC8",
    "#C5E1A5",
    "#AED581",
    "#9CCC65",
    "#8BC34A",
    "#7CB342",
    "#689F38",
    "#558B2F",
    "#33691E",
  ],
  Array [
    "#F9FBE7",
    "#F0F4C3",
    "#E5EE9C",
    "#DBE775",
    "#D3E157",
    "#CCDC39",
    "#BFCA33",
    "#AEB42B",
    "#9D9D24",
    "#817717",
  ],
  Array [
    "#FFFDE7",
    "#FEF8C3",
    "#FDF39B",
    "#FBEE73",
    "#F8E852",
    "#FFEB3B",
    "#FDD835",
    "#FBC02D",
    "#F9A825",
    "#F57F16",
  ],
  Array [
    "#FFF8E1",
    "#FFECB3",
    "#FFE082",
    "#FFD54F",
    "#FFCA28",
    "#FFC107",
    "#FFB300",
    "#FFA000",
    "#FF8F00",
    "#FF6F00",
  ],
  Array [
    "#FFF3E0",
    "#FFE0B2",
    "#FFCD80",
    "#FFB84D",
    "#FFA826",
    "#FF9900",
    "#FB8D00",
    "#F57D00",
    "#EF6D00",
    "#E65200",
  ],
  Array [
    "#FBE9E7",
    "#FFCCBC",
    "#FFAB91",
    "#FF8965",
    "#FF6F43",
    "#FF5622",
    "#F4501E",
    "#E64919",
    "#D84215",
    "#BF350C",
  ],
  Array [
    "#EFEBE9",
    "#D7CCC8",
    "#BCAAA4",
    "#A1887F",
    "#8D6E63",
    "#795548",
    "#6D4C41",
    "#5D4037",
    "#4E342E",
    "#3E2723",
  ],
  Array [
    "#FAFAFA",
    "#F4F4F4",
    "#EEEEEE",
    "#E0E0E0",
    "#BDBDBD",
    "#9E9E9E",
    "#757575",
    "#616161",
    "#424242",
    "#212121",
  ],
  Array [
    "#ECEFF1",
    "#CFD8DC",
    "#B0BEC5",
    "#90A4AE",
    "#78909C",
    "#607D8B",
    "#546E7A",
    "#455A64",
    "#37474F",
    "#263238",
  ],
]
`;
